types/store.go

```diff
@@ -222,3 +230,29 @@ func (key *KVStoreKey) Name() string {
 func (key *KVStoreKey) String() string {
 	return fmt.Sprintf("KVStoreKey{%p, %s}", key, key.name)
 }
+
+// PrefixEndBytes returns the []byte that would end a
+// range query for all []byte with a certain prefix
+// Deals with last byte of prefix being FF without overflowing
+func PrefixEndBytes(prefix []byte) []byte {
```

Comment:
We should write a test for this function (and more comments) - what happens if `prefix = []byte{0xFF}`

Reply:
It's tested in the SubspaceIterator tests

Reply:
https://github.com/cosmos/cosmos-sdk/blob/sunny/IAVLsubspace/store/iavlstore_test.go#L151

Reply:
I also have a test for it in the PR to tmlibs.  Should I just readd that here?
https://github.com/tendermint/tmlibs/blob/PrefixEndBytes/common/byteslice_test.go#L9

Reply:
Well this should refer to the tmlibs function if it's going to be in tmlibs in which case the tmlibs test is fine?

Reply:
Added

Reply:
{[]byte{byte(255)} ?

Reply:
Yeah how are we handling it that case, that test isn't in `TestPrefixEndBytes`

Reply:
You return nil.  That's essentially the same case as `[]byte{byte(255), byte(255), byte(255)}` but sure, I'll add it as well.